bindgen/clang.rs

```diff
@@ -1868,6 +1868,23 @@ impl TranslationUnit {
         }
     }
 
+    /// Save a translation unit to the given file.
+    pub(crate) fn save(&mut self, file: &str) -> bool {
```

Comment:
Yeah I tried to keep it consistent with the other APIs so I think `&str` is best. As for `Result`, I don't quite know what we would do with the error. One of the things I've noticed in the code I've seen is that most things either return an `Option` or nothing and panic. Maybe @emilio has some preferences here, but I want to stick to the existing coding style.

Reply:
Yeah, `&CStr` would save an allocation, but consistency may be a reason to prefer `&str`.

For `Result`, in this case I was simply thinking of `Result<(), ()>` -- I saw that there were some `-> bool`s around, but they seemed to be `is_xyz()` kind of methods where it makes more sense, and there are some `Result<T, ()>`s as well (and with other error types of course).

Even if we need to ignore the error explicitly, I think it is better to use `Result` than to use a non-`must_use` type instead to avoid that, but of course, please take it with a grain of salt since I don't know the codebase.

Reply:
I'd return `Result<(), CXSaveError>` to at least express the intent of returning an error that could be displayed properly in the future if this were to fail.